components/global/rg.tf

```diff
@@ -1,3 +1,13 @@
 data "azurerm_resource_group" "main" {
   name = "lz-${var.env}-rg"
+}
+
+data "azurerm_resource_group" "key_vault" {
+  name = var.env == "perftest" || var.env == "aat" || var.env == "preview" ? "core-infra-${var.subscription}-rg" : "core-infra-${var.env}-rg"
+}
+
+data "azurerm_log_analytics_workspace" "log" {
+  provider            = azurerm.loganalytics
+  name                = local.log_analytics_workspace[[for x in keys(local.log_analytics_env_mapping) : x if contains(local.log_analytics_env_mapping[x], var.env)][0]].name
```

Comment:
you can use the log analytics module for this?

https://github.com/hmcts/terraform-module-log-analytics-workspace-id/

would be nicer